File: libNative/src/main/java/com/munch/lib/nativelib/helper/ViewHelper.java
package com.munch.lib.nativelib.helper;

import android.support.annotation.NonNull;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Munch on 2018/12/13.
 */
public class ViewHelper {

    /**
     * @param l 当检查到空值时的回调
     * @return 按照传入TextView的顺序返回其值的集合，当有空值时返回空数组
     */
    @NonNull
    public static String[] checkEmptyAndGetValue2(OnCheckListener l, TextView... views) {
        int len = views.length;
        String[] strings = new String[len];
        TextView viewTemp;
        for (int i = 0; i < len; i++) {
            viewTemp = views[i];
            String val = viewTemp.getText().toString();
            if (StringHelper.isEmpty(val)) {
                l.onEmpty(viewTemp);
                return new String[0];
            } else {
                strings[i] = val;
            }
        }
        return strings;
    }

    /**
     * @param l 当检查到空值时的回调，根据返回值决定是否继续检查，最后返回值的集合
     */
    public static void checkEmptyAndGetValue(OnCheckListener l, TextView... views) {
        List<String> strings = new ArrayList<>(views.length);
        for (TextView view : views) {
            String val = view.getText().toString();
            if (StringHelper.isEmpty(val) && !l.onEmpty(view)) {
                return;
            } else {
                strings.add(val);
            }
        }
        l.onVals(strings);
    }

    public interface OnCheckListener {
        /**
         * 当检查到空值时的处理回调
         *
         * @return 是否继续检测
         */
        boolean onEmpty(TextView view);

        void onVals(List<String> vals);
    }
}

File: libnative/src/main/java/com/munhc/lib/libnative/helper/ViewHelper.kt
package com.munhc.lib.libnative.helper

import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.TextView
import androidx.core.view.children
import com.munhc.lib.libnative.excetion.MethodEcxception

/**
 * Created by Munch on 2019/7/15 13:50
 */
object ViewHelper {

    fun setNonInput(editText: EditText) {
        editText.keyListener = null
        editText.isFocusable = false
    }

    fun getParams(view: View): ViewGroup.LayoutParams {
        return view.layoutParams ?: ViewGroup.LayoutParams(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.WRAP_CONTENT
        )
    }

    fun clickItem(vg: ViewGroup, listener: View.OnClickListener, vararg clazz: Class<in View>) {
        vg.children.forEachIndexed { index, view ->
            if (clazz.isNotEmpty()) {
                clazz.forEach {
                    if (it.isAssignableFrom(view::class.java)) {
                        view.tag = index
                        view.setOnClickListener(listener)
                    }
                }
            } else {
                view.tag = index
                view.setOnClickListener(listener)
            }
        }

    }

    fun setTextViewNonVal(views: Array<TextView>, strs: Array<String?>) {
        if (views.size != strs.size) {
            throw MethodEcxception.wrongParameter()
        }
        views.forEachIndexed { index, textView ->
            textView.text = strs[index] ?: ""
        }
    }

    fun checkTextEmpry(listener: OnCheckTextViewListener, vararg views: TextView) {
        if (views.isEmpty()) {
            return
        }
        val array = ArrayList<String>(views.size)
        views.forEach {
            val str = it.text.toString()
            if (str.isEmpty() && !listener.onCheckEmpty(it)) {
                return
            }
            array.add(str)
        }
        listener.onResult(array)
    }

    interface OnCheckTextViewListener {

        /**
         * 当检查到view的text为空时回调
         *
         * @param view 数据为空的Textview或其子类
         * @return 是否继续检查剩余的View，true则继续
         */
        fun onCheckEmpty(view: TextView): Boolean

        /**
         * 当检查完所有view都不为空或者有为空但返回true时回调
         * 注意vals的长度
         *
         * @param vals 按顺序获取的Textview或其子类的值的集合
         */
        fun onResult(vals: List<String>)
    }

}